File: java/src/main/java/org/endeavourhealth/library/model/iml/Page.java
package org.endeavourhealth.library.model.iml;

public class Page {
  private Integer pageNumber;
  private Integer pageSize;
  private Integer offset;

  public Integer getOffset() {
    return offset;
  }

  public Page setOffset(Integer offset) {
    this.offset = offset;
    return this;
  }

  public Integer getPageNumber() {
    return pageNumber;
  }

  public Page setPageNumber(Integer pageNumber) {
    this.pageNumber = pageNumber;
    return this;
  }

  public Integer getPageSize() {
    return pageSize;
  }

  public Page setPageSize(Integer pageSize) {
    this.pageSize = pageSize;
    return this;
  }
}

File: src/interfaces/Page.ts
import z from "zod";

// export interface Page {
//   pageNumber?: number;
//   pageSize?: number;
//   offset?: number;
// }

export const PageSchema = z.strictObject({
  pageNumber: z.number().default(1),
  pageSize: z.number().default(20),
  offset: z.number().default(0)
});

export type Page = z.output<typeof PageSchema>;
